### PR TITLE
AzureDataShare - New version 2020-09-01

### PR DESCRIPTION
**Pull request by vnvaditya**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by vnvaditya** [comment]
Adding a new version 2020-09-01. Current PR doesn't reflect the changes. This PR is just a copy of previous stable swagger.

Once this is approved and merged, we will follow up with new PR with actual changes.

**Comment by varunkumta** [comment]
Thank you for making the first commit as the previous api-version's content! Since the 2nd commit is not yet made, I am removing the WaitForARMFeedback label. When you are ready with the new api-version content, please check the first checkbox under "ARM API Review Checklist" in the PR description and the label will be added back

**Comment by vnvaditya** [comment]
After the latest push, there are no breaking changes. 

Commit 1(44f2d328385582e5d65be6a44119e9a23c11eb83)  is a simple copy of previous stable version. 

To see the diff for the latest version, please compare commit 1 vs commit 5

**Comment by varunkumta** [comment]
>               "$ref": "#/definitions/ProviderShareSubscription"

this is modeled as a post, but the request/response of the post (and response of the get) is the full resource schema. Any reason this wasn't modeled as a PUT?

---
Refers to: specification/datashare/resource-manager/Microsoft.DataShare/stable/2020-09-01/DataShare.json:1490 in 3918791. [](commit_id = 391879154016edaef007d5701edfe141546435c5, deletion_comment = False)

**Comment by varunkumta** [comment]
>         "description": "Create a synchronizationSetting",

are you not supporting update any more?

---
Refers to: specification/datashare/resource-manager/Microsoft.DataShare/stable/2020-09-01/DataShare.json:2711 in 3918791. [](commit_id = 391879154016edaef007d5701edfe141546435c5, deletion_comment = False)

**Comment by varunkumta** [comment]
>     "SynapseWorkspaceSqlPoolTableDataSet": {

where is this ref'ed?

---
Refers to: specification/datashare/resource-manager/Microsoft.DataShare/stable/2020-09-01/DataShare.json:6349 in 3918791. [](commit_id = 391879154016edaef007d5701edfe141546435c5, deletion_comment = False)

**Comment by varunkumta** [comment]
>     "SynapseWorkspaceSqlPoolTableDataSetMapping": {

this too

---
Refers to: specification/datashare/resource-manager/Microsoft.DataShare/stable/2020-09-01/DataShare.json:6370 in 3918791. [](commit_id = 391879154016edaef007d5701edfe141546435c5, deletion_comment = False)

**Comment by varunkumta** [comment]
>             "Ok",

use standard values for status like Succeeded and Failed, instead of Ok and Broken

---
Refers to: specification/datashare/resource-manager/Microsoft.DataShare/stable/2020-09-01/DataShare.json:6406 in 3918791. [](commit_id = 391879154016edaef007d5701edfe141546435c5, deletion_comment = False)

**Comment by varunkumta** [comment]
>           "description": "Provisioning state of the data set mapping.",

is this different from the property above?

---
Refers to: specification/datashare/resource-manager/Microsoft.DataShare/stable/2020-09-01/DataShare.json:6417 in 3918791. [](commit_id = 391879154016edaef007d5701edfe141546435c5, deletion_comment = False)

File: specification/datashare/resource-manager/Microsoft.DataShare/stable/2020-09-01/examples/DataSetMappings_SynapseWorkspaceSqlPoolTable_Create.json
Comment (varunkumta) [inline specification/datashare/resource-manager/Microsoft.DataShare/stable/2020-09-01/examples/DataSetMappings_SynapseWorkspaceSqlPoolTable_Create.json:36]: >icrosoft.DataShare/DataSetMapping [](start = 18, length = 33)

this should be the fully qualified resource type i.e. in this case - Microsoft.DataShare/accounts/shareSubscriptions/dataSetMappings .. I realize it was done this way in the previous api-version too .. please fix it for this api-version

**Review by varunkumta — CHANGES_REQUESTED** [review]
please take a look at review comments

**Review by varunkumta — COMMENTED** [review]
not blocking review so next reviewer can sign off

**Review by varunkumta — COMMENTED** [review]
not blocking review so next reviewer can sign off

**Comment by ahermas** [comment]
> > ```
> >     "description": "Create a synchronizationSetting",
> > ```
> 
> are you not supporting update any more?
> 
> Refers to: specification/datashare/resource-manager/Microsoft.DataShare/stable/2020-09-01/DataShare.json:2711 in [3918791](https://github.com/Azure/azure-rest-api-specs/commit/391879154016edaef007d5701edfe141546435c5). [](commit_id = [3918791](https://github.com/Azure/azure-rest-api-specs/commit/391879154016edaef007d5701edfe141546435c5), deletion_comment = False)

Regarding your comment about "description": "Create a synchronizationSetting", there isn't any change btw version 2019-011-01 and the newly introduced 2020-09-01.

**Review by varunkumta — APPROVED** [review]
approving to unblock review. However, please have the current ARM api reviewer look at latest changes - and they will add the ARMSignedOff label to actually indicate sign-off. Tagging @j5lim @pilor who are api reviewers for this week

**Comment by ahermas** [comment]
> > ```
> >           "$ref": "#/definitions/ProviderShareSubscription"
> > ```
> 
> this is modeled as a post, but the request/response of the post (and response of the get) is the full resource schema. Any reason this wasn't modeled as a PUT?
> 
> Refers to: specification/datashare/resource-manager/Microsoft.DataShare/stable/2020-09-01/DataShare.json:1490 in [3918791](https://github.com/Azure/azure-rest-api-specs/commit/391879154016edaef007d5701edfe141546435c5). [](commit_id = [3918791](https://github.com/Azure/azure-rest-api-specs/commit/391879154016edaef007d5701edfe141546435c5), deletion_comment = False)

This is a view of a resource on the consumer side that the provider can act on (specifics of the data share). The provider cannot create the resource themselves, hence there is no PUT and is modeled as a POST.

**Comment by ahermas** [comment]
> > ```
> >         "Ok",
> > ```
> 
> use standard values for status like Succeeded and Failed, instead of Ok and Broken
> 
> Refers to: specification/datashare/resource-manager/Microsoft.DataShare/stable/2020-09-01/DataShare.json:6406 in [3918791](https://github.com/Azure/azure-rest-api-specs/commit/391879154016edaef007d5701edfe141546435c5). [](commit_id = [3918791](https://github.com/Azure/azure-rest-api-specs/commit/391879154016edaef007d5701edfe141546435c5), deletion_comment = False)

This is not provisioning status, this field is again specifics for data share implementation and we are using Ok and broken. DataSetMapping also has a provisioning state which follows the regular convention.

**Comment by ahermas** [comment]
> > ```
> > "SynapseWorkspaceSqlPoolTableDataSet": {
> > ```
> 
> where is this ref'ed?
> 
> Refers to: specification/datashare/resource-manager/Microsoft.DataShare/stable/2020-09-01/DataShare.json:6349 in [3918791](https://github.com/Azure/azure-rest-api-specs/commit/391879154016edaef007d5701edfe141546435c5). [](commit_id = [3918791](https://github.com/Azure/azure-rest-api-specs/commit/391879154016edaef007d5701edfe141546435c5), deletion_comment = False)

This uses the kind discriminator on data set . So essentially this is a type of data set

**Comment by ahermas** [comment]
> > ```
> > "SynapseWorkspaceSqlPoolTableDataSetMapping": {
> > ```
> 
> this too
> 
> Refers to: specification/datashare/resource-manager/Microsoft.DataShare/stable/2020-09-01/DataShare.json:6370 in [3918791](https://github.com/Azure/azure-rest-api-specs/commit/391879154016edaef007d5701edfe141546435c5). [](commit_id = [3918791](https://github.com/Azure/azure-rest-api-specs/commit/391879154016edaef007d5701edfe141546435c5), deletion_comment = False)

This uses the kind discriminator on data set mapping. So essentially this is a type of data set mapping

**Comment by ahermas** [comment]
> > ```
> >       "description": "Provisioning state of the data set mapping.",
> > ```
> 
> is this different from the property above?
> 
> Refers to: specification/datashare/resource-manager/Microsoft.DataShare/stable/2020-09-01/DataShare.json:6417 in [3918791](https://github.com/Azure/azure-rest-api-specs/commit/391879154016edaef007d5701edfe141546435c5). [](commit_id = [3918791](https://github.com/Azure/azure-rest-api-specs/commit/391879154016edaef007d5701edfe141546435c5), deletion_comment = False)

Yes it's different.  The provisioning state of the DataSetMapping follows the regular convention that you referred to earlier with values  like Succeeded and Failed

File: specification/datashare/resource-manager/Microsoft.DataShare/stable/2020-09-01/DataShare.json
Comment (pilor) [inline specification/datashare/resource-manager/Microsoft.DataShare/stable/2020-09-01/DataShare.json:1490]: Please modify the examples to show what this parameter is used for in adjust and reinstate.  Its not clear when/how this would be used if it is not required.  Its also odd that it takes the entire providerShareSubscription when it is only changing 1 property.

File: specification/datashare/resource-manager/Microsoft.DataShare/stable/2020-09-01/examples/DataSetMappings_SynapseWorkspaceSqlPoolTable_Create.json
Comment (ahermas) [reply specification/datashare/resource-manager/Microsoft.DataShare/stable/2020-09-01/examples/DataSetMappings_SynapseWorkspaceSqlPoolTable_Create.json:36]: @pilor The fully qualified resource type change is recent and we're also wondering if the resource for the existing api versions should be updated too?

Comment (pilor) [reply specification/datashare/resource-manager/Microsoft.DataShare/stable/2020-09-01/examples/DataSetMappings_SynapseWorkspaceSqlPoolTable_Create.json:36]: The resource type that corresponds with that resource ID (and the type that RBAC will be checked on) is Microsoft.DataShare/accounts/shareSubscriptions/dataSetMappings.  Anything else is incorrect

Comment (sarajang) [reply specification/datashare/resource-manager/Microsoft.DataShare/stable/2020-09-01/examples/DataSetMappings_SynapseWorkspaceSqlPoolTable_Create.json]: @pilor updated.

File: specification/datashare/resource-manager/Microsoft.DataShare/stable/2020-09-01/DataShare.json
Comment (sarajang) [reply specification/datashare/resource-manager/Microsoft.DataShare/stable/2020-09-01/DataShare.json:1490]: @pilor  updated.

**Comment by sarajang** [comment]
@pilor can you please review the updated changes?

**Comment by ahermas** [comment]
> @pilor can you please review the updated changes?

@j5lim @pilor  Could you both please review the changes? Thanks!

**Review by j5lim — CHANGES_REQUESTED** [review]

File: specification/datashare/resource-manager/Microsoft.DataShare/stable/2020-09-01/DataShare.json
Comment (jorgecotillo) [inline specification/datashare/resource-manager/Microsoft.DataShare/stable/2020-09-01/DataShare.json:5137]: What is this value for? I don't see an example using this value, is this meant to be a boolean?

File: specification/datashare/resource-manager/Microsoft.DataShare/stable/2020-09-01/DataShare.json
Comment (jorgecotillo) [inline specification/datashare/resource-manager/Microsoft.DataShare/stable/2020-09-01/DataShare.json:6353]: Is this object used in any request or response?

File: specification/datashare/resource-manager/Microsoft.DataShare/stable/2020-09-01/DataShare.json
Comment (jorgecotillo) [inline specification/datashare/resource-manager/Microsoft.DataShare/stable/2020-09-01/DataShare.json:6403]: If this value is set by your service, mark it as readOnly

File: specification/datashare/resource-manager/Microsoft.DataShare/stable/2020-09-01/DataShare.json
Comment (jorgecotillo) [inline specification/datashare/resource-manager/Microsoft.DataShare/stable/2020-09-01/DataShare.json:6436]: Please review all the new properties and mark them as readOnly if the value is meant to be set by your service only.

**Review by jorgecotillo — CHANGES_REQUESTED** [review]

**Comment by sarajang** [comment]
https://github.com/Azure/azure-rest-api-specs/pull/12745 Started a new PR.

**Pull request closed without merging**
